### PR TITLE
RMNhome reader lifecycle

### PR DESCRIPTION
**Pull request by 0xnogo**
Starting and Cancelling the RMNHome Reader in the Commit Plugin

### CODE REVIEW

File: commit/plugin.go [hunk 1]
```diff
@@ -79,6 +80,18 @@ func NewPlugin(
...
 
+	if offchainCfg.RMNEnabled {
+		if err := rmnHomeReader.Ready(); err == nil {
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			if err := rmnHomeReader.Start(ctx); err != nil {
+				lggr.Errorw("Failed to start RMNHome reader", "err", err)
+			}
+		} else {
+			lggr.Errorw("Failed to initialize RMNHome reader", "err", err)
```
Comment (makramkd) [inline commit/plugin.go:91]: If RMNHome reader isn't working and RMN is enabled, should we even start the plugin?

File: commit/plugin.go [hunk 2]
```diff
@@ -79,6 +80,18 @@ func NewPlugin(
 		offchainCfg.MaxMerkleTreeSize = merklemulti.MaxNumberTreeLeaves
 	}
 
+	if offchainCfg.RMNEnabled {
+		if err := rmnHomeReader.Ready(); err == nil {
```
Comment (makramkd) [inline commit/plugin.go:84]: What does `Ready` do exactly?

Comment (0xnogo) [reply commit/plugin.go:84]: Checks that the service has not been started yet, which in this case should not be happening.

File: commit/plugin.go (hunk 1 above)
Comment (0xnogo) [reply commit/plugin.go]: As discussed offline, I moved the init logic in the factory to let the libocr retry when `NewReportingPlugin` errored

File: commit/factory.go [hunk 3]
```diff
@@ -130,6 +130,15 @@ func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types
 			p.lggr,
 			100*time.Millisecond,
 		)
+
+		if err := rmnHomeReader.Ready(); err != nil {
+			return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("failed to initialize RMNHome reader: %w", err)
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (makramkd) [inline commit/factory.go:137]: We get a ctx in the method already, can we just use that?

Comment (0xnogo) [reply commit/factory.go:137]: My goal is to create a context with a timeout, but I guess we could create a child context

File: commit/factory.go [hunk 4]
```diff
@@ -130,6 +130,15 @@ func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types
 			p.lggr,
 			100*time.Millisecond,
 		)
+
+		if err := rmnHomeReader.Ready(); err != nil {
+			return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("failed to initialize RMNHome reader: %w", err)
+		}
+		childCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+		if err := rmnHomeReader.Start(childCtx); err != nil {
```
Comment (makramkd) [inline commit/factory.go:139]: Why not just rmnHomeReader.Start(ctx)? I believe that if the parent context cancels then this child context will cancel even if you have a longer timeout than the parent context.

Comment (0xnogo) [reply commit/factory.go]: done

File: commit/plugin.go [hunk 5]
```diff
@@ -346,6 +346,22 @@ func (p *Plugin) Outcome(
...
+			}
+		} else {
+			p.lggr.Warn("RMNHome reader was nil during Close")
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("errors during Close: %v", errs)
+	}
 	return nil
```
Comment (makramkd) [inline commit/plugin.go:365]: Use [errors.Join](https://pkg.go.dev/errors#Join) instead:

```suggestion
	return errors.Join(errs)
```

**Review by makramkd — APPROVED** [review]

**Pull request merged**